common/lib/xmodule/xmodule/library_tools.py

```diff
@@ -96,6 +96,17 @@ def _filter_child(self, usage_key, capa_type):
         assert isinstance(descriptor, CapaDescriptor)
         return capa_type in descriptor.problem_types
 
+    def can_use_library_content(self, course_id):
+        """
+        Determines whether a modulestore holding a course_id supports libraries.
+        """
+        store = self.store
+        # Skip MixedModuleStore abstraction.
+        if hasattr(store, '_get_modulestore_for_courseid'):
+            # pylint: disable=protected-access
+            store = store._get_modulestore_for_courselike(course_id)
+        return hasattr(store, "copy_from_template")
```

Comment:
@Kelketek from one point of view, I don't like accessing private methods and breaking abstractions. From other - probably it's the least involving way to do it. I'll list a couple of alternatives so we could discuss them and decide if more complex solution worth it.

Ok, so the advantages of current approach is it's duck-typing'ish - it checks if store supports *method* that we need, noot it's type, allowing modulestores to automatically pass this check the moment `copy-from_template` is implemented on them. Disadvantages are: (1) usage of private methods and (2) breach of "mixed" modulestore abstraction.

* Use `get_modulestore_type(course_id)` and check if it's `ModuleStoreEnum.Type.split` - does not use private methods and does not break "mixed" abstraction, but asserts on the type of store, not the methods it supports. So it's the reverse of current approach.
* Push down `_get_modulestore_or_courselike` into actual modulestores, make it public; than just always do `hasattr(get_modulestore_or_courselike(course_id), "copy_from_template")
* Push `_verify_modulestore_supports` into actual modulestores, make it public, than just do `verify_modulestore_supports(course_id, 'copy_from_template')

The last two still do duck-typing'ish check, does not break "mixed" abstraction and not use private methods, but will change public interface and require small changes for all modulestores. However, `verify_modulestore_supports` would be just `hasattr(self, 'method)` for most (if not all) of stores, so could be placed at the very top of modulestore hierarchy.

So, some input from core devs on idea to make `verify_modulestore_supports` public and push it to all modulestores. I believe @dmitchell and @cpennington were the reviewers for other ContentLibraries modulestore-related stuff, if I'm not mistaking. So, could you (@dmitchell and/or @cpennington) please take a look on it when you have time, or redirect the request to someone else if I was wrong.